analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant_RoslynCfg.cs

```diff
@@ -0,0 +1,161 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.CFG.Helpers;
+using SonarAnalyzer.CFG.Roslyn;
+using SonarAnalyzer.Extensions;
+using StyleCop.Analyzers.Lightup;
+using SymbolWithInitializer = System.Collections.Generic.KeyValuePair<Microsoft.CodeAnalysis.ISymbol, Microsoft.CodeAnalysis.CSharp.Syntax.EqualsValueClauseSyntax>;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    public sealed partial class MemberInitializerRedundant
+    {
+        private class MemberInitializerRedundancyCheckerRoslyn : CfgAllPathValidator
+        {
+            private readonly IsValidMatcher isValidMatcher;
+            private readonly IsInvalidMatcher isInvalidMatcher;
+
+            public MemberInitializerRedundancyCheckerRoslyn(ControlFlowGraph cfg, ISymbol memberToCheck, SemanticModel semanticModel) : base(cfg)
+            {
+                var redundancyChecker = new RedundancyChecker(memberToCheck, semanticModel);
+                this.isValidMatcher = new IsValidMatcher(redundancyChecker);
+                this.isInvalidMatcher = new IsInvalidMatcher(redundancyChecker);
+            }
+
+            // Returns true if the block contains assignment before access
+            protected override bool IsValid(BasicBlock block) =>
+                isValidMatcher.Matches(block);
+
+            // Returns true if the block contains access before assignment
+            protected override bool IsInvalid(BasicBlock block) =>
+                isInvalidMatcher.Matches(block);
+
+            private abstract class RedundancyMatcherBase : OperationFinder<bool>
```

Comment:
Remove `Redundancy` because the inherited classes don't have it in the name.
`Finder` because it's a finder.
Only `FinderBase`, because it's private class and it doesn't need a long name.

```suggestion
            private abstract class FinderBase : OperationFinder<bool>
```